src/components/delete-trace.tsx

```diff
@@ -78,3 +78,34 @@ export function DeleteTrace({
     </Popover>
   );
 }
+
+export function DeleteTraceMultiSelectAction({
+  traceIds,
+  projectId,
+}: {
+  traceIds: string[];
+  projectId: string;
+}) {
+  const utils = api.useUtils();
+
+  const hasAccess = useHasAccess({ projectId, scope: "traces:delete" });
+
+  const mutDeleteTraces = api.traces.deleteMany.useMutation({
+    onSuccess: () => void utils.traces.invalidate(),
+  });
+
+  if (!hasAccess) {
+    return null;
+  }
+
+  return (
+    <Button
```

Comment:
I'd stick to the DropdownMenuItem (shadcn/ui) aesthetic instead of inserting a button into the DropdownMenuItem.
Reference: https://ui.shadcn.com/docs/components/dropdown-menu